merlin/study/script_adapter.py

```diff
@@ -79,22 +80,23 @@ def __init__(self, **kwargs):
             "lsf": "",
         }
 
-        self._unsupported = {
+        self._unsupported: Set[str] = {
```

Comment:
now that i look at this. why is this a set and the other one (new_unsupported) a list?

Reply:
Mm, I don't know ¯\\_(ツ)_/¯. AFAIS `new_unsupported` is only used in the constructor to build a new `_unsupported` set by union'ing its list of strs and then it's dumped. Maybe `_unsupported` needs to be guaranteed collision free somewhere else and that's why it's a Set (it's a Set because that's what it is in the parent class from Maestro), and `new_unsupported` is a List because it doesn't matter after construction?

I tend to make these sorts of symbols Enums rather than handle Lists/Sets of raw strings, although being fed these data types by inheritance from Maestro kind of ties hands.

Reply:
My confusion was why MerlinLSFScriptAdapter was dumping the inherited attributes it initializes in the parent and overwriting with entirely new, largely parallel data rather than updating the homologous data structs the parent already initialized like MerlinSlurmScriptAdapter does. It's nice MerlinLSFScriptAdapter makes it explicit what keys you could grab in Merlin from `_unsupported`, especially given its parent is a library that has to be accessed through linking/the venv, but, we might want to pick one or the other.

Reply:
The set was used to eliminate duplicates from the lists. The various unsupported lists were created to work around maestro issues. This can be refactored to create a generic parallel launcher with external configuration. This was one of the feature proposals near the end of the project.